tracer/src/Datadog.Tracer.Native/tracer_tokens.cpp

```diff
@@ -152,13 +153,33 @@ HRESULT TracerTokens::EnsureBaseCalltargetTokens()
     {
         return hr;
     }
+
+    
+
     // *** Ensure Datadog.Trace.ClrProfiler.CallTarget.CallTargetBubbleUpException type ref, might not be available if tracer version is < 2.22
-    if (bubbleUpExceptionTypeRef == mdTypeRefNil)
+    if (trace::profiler->IsCallTargetBubbleUpExceptionTypeAvailable() && bubbleUpExceptionTypeRef == mdTypeRefNil)
     {
         const ModuleMetadata* module_metadata = GetMetadata();
-        module_metadata->metadata_emit->DefineTypeRefByName(profilerAssemblyRef,
+        const auto defined_call_target_bubble_up_exception_type = module_metadata->metadata_emit->DefineTypeRefByName(profilerAssemblyRef,
                                                             calltargetbubbleexception_tracer_type_name.c_str(),
                                                             &bubbleUpExceptionTypeRef);
+        if (SUCCEEDED(defined_call_target_bubble_up_exception_type) && trace::profiler->IsCallTargetBubbleUpFunctionAvailable() && bubbleUpExceptionFunctionRef == mdMemberRefNil)
+        {
+            // now reference the method IsCallTargetBubbleUpException in the type's reference
+            COR_SIGNATURE createInstanceSig[32];
+            COR_SIGNATURE* sigBuilder = createInstanceSig;
+            sigBuilder += CorSigCompressData(IMAGE_CEE_CS_CALLCONV_DEFAULT, sigBuilder); // static
+            sigBuilder += CorSigCompressData(1, sigBuilder); // arguments count
+            sigBuilder += CorSigCompressElementType(ELEMENT_TYPE_BOOLEAN, sigBuilder);     // return type
+            sigBuilder += CorSigCompressElementType(ELEMENT_TYPE_CLASS, sigBuilder); // argument type
+            sigBuilder += CorSigCompressToken(exTypeRef, sigBuilder);
+            
+            const auto defined_call_target_bubble_up_exception_function = module_metadata->metadata_emit->DefineMemberRef(
```

Comment:
What happens if `defined_call_target_bubble_up_exception_function` HRESULT returns FAIL? this result is not using anywhere... if the code is prepared to handle `bubbleUpExceptionFunctionRef == mdMemberRefNil` later maybe you should force that on the FAIL case.

Reply:
added in [0721e36](https://github.com/DataDog/dd-trace-dotnet/pull/5034/commits/0721e36e224c9d302ddc56c42a4b61bf41f954c5)